### PR TITLE
Fix C# reference TOC for offline book

### PR DESCRIPTION
**Pull request by gewarren**
For the offline book, the first node must be unnested or a parent of the other nodes. LMK if this looks okay to you @BillWagner.

([Hide whitespace changes](https://github.com/dotnet/docs/pull/40794/files?diff=unified&w=1).)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/tools/dotnet-new-sdk-templates.md](https://github.com/dotnet/docs/blob/c6d212cf6fc25064d175793e236c554e7b14487b/docs/core/tools/dotnet-new-sdk-templates.md) | [.NET default templates for dotnet new](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-new-sdk-templates?branch=pr-en-us-40794) |
| [docs/csharp/language-reference/compiler-options/language.md](https://github.com/dotnet/docs/blob/c6d212cf6fc25064d175793e236c554e7b14487b/docs/csharp/language-reference/compiler-options/language.md) | [C# Compiler Options for language feature rules](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/language?branch=pr-en-us-40794) |
| [docs/csharp/language-reference/configure-language-version.md](https://github.com/dotnet/docs/blob/c6d212cf6fc25064d175793e236c554e7b14487b/docs/csharp/language-reference/configure-language-version.md) | [Configure C# language version - C# Guide](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version?branch=pr-en-us-40794) |
| [docs/csharp/language-reference/language-versioning.md](https://github.com/dotnet/docs/blob/c6d212cf6fc25064d175793e236c554e7b14487b/docs/csharp/language-reference/language-versioning.md) | [C# language versioning](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-versioning?branch=pr-en-us-40794) |
| [docs/csharp/misc/cs1617.md](https://github.com/dotnet/docs/blob/c6d212cf6fc25064d175793e236c554e7b14487b/docs/csharp/misc/cs1617.md) | [docs/csharp/misc/cs1617](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs1617?branch=pr-en-us-40794) |

### CODE REVIEW

**Comment by BillWagner** [comment]
This looks great. Do we need to make a similar change to the related TOCs? C# Guide, and spec?

**Comment by BillWagner** [comment]
Something in the build clearly didn't work. The TOC for any reference page is missing a lot of nodes.

**Comment by gewarren** [comment]
> This looks great. Do we need to make a similar change to the related TOCs? C# Guide, and spec?

No, the other two are working fine.

File: docs/csharp/language-reference/toc.yml
Comment (BillWagner) [inline docs/csharp/language-reference/toc.yml:5]: The other 2 TOCs have only one "root" node, with an "items node that have every other item under it. It would look something like this here. (Not a suggestion, because I can't create a suggestion on deleted lines:

```yml
items:
- name C# language reference
  href: ./index.yml
  items:
  -name: Configure language version
    href: ./configure-language-version.md
  -name: Types
    items:
    # remaining items
```

**Review by BillWagner — COMMENTED** [review]
This looks like the differences I can find between this TOC and the others.

File: docs/csharp/language-reference/toc.yml
Comment (gewarren) [reply docs/csharp/language-reference/toc.yml:2348]: The reason I removed them is because all the other nodes are really part of "C# reference documentation", but these two aren't, and yet they're at the same level.

Comment (BillWagner) [reply docs/csharp/language-reference/toc.yml:2348]: That's why I put them at the root level (on all three C# based TOCs)

It's not critical, but it was a consistent navigation element.

**Review by BillWagner — APPROVED** [review]
This is good to merge once we get the navigation working.

**Comment by gewarren** [comment]
Latest theory: This only doesn't work on the review site, so I'll merge this and then into live and see.

**Comment by mdell-seradex** [comment]
Why did this change remove a lot of information including warnings from [Configure C# language version](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#edit-the-project-file)?
It used to say things like `Choosing a language version newer than the default can cause hard to diagnose compile-time and runtime errors.`
Note that [C# Compiler Options for language feature rules](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/language#langversion) implies that this may still be the case since it says `The latest value is generally not recommended. With latest, the compiler enables the latest features, even if those features depend on updates not included in the configured target framework.`

**Comment by gewarren** [comment]
> Can someone please provide some information as to why this change removed a lot of information including warnings from [Configure C# language version](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#edit-the-project-file). It used to say things like `Choosing a language version newer than the default can cause hard to diagnose compile-time and runtime errors.`

This file was split into two files - the info you're looking for is in this file now: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-versioning.

**Comment by mdell-seradex** [comment]
@gewarren, I believe I looked at that `language versioning` page before, but did review it again now.  I do not see anywhere that it includes the warnings (or something equivalent) that were removed from the `Configure C# language version` page, such as `Choosing a language version newer than the default can cause hard to diagnose compile-time and runtime errors.`  If that no longer applies, it would be great if that was noted somewhere, like in a footnote.
Based on the note in the `C# Compiler Options for language feature rules` page (`The latest value is generally not recommended. With latest, the compiler enables the latest features, even if those features depend on updates not included in the configured target framework`), I would say that this still applies.  I would even venture to guess based on the now removed notes from `Configure C# language version` that this is true any time a version of C# is targetted that is higher than the default for the version of .NET that a project targets.

Based on that, I feel that the recommendation given to me in the referenced `Virtual indentation` issue to target C# 11 for a .NET Framework project is a bad recommendation.
I would love to hear your thoughts on this.

Thank you.

**Comment by gewarren** [comment]
@mdell-seradex That warning is the last sentence of the first paragraph:

image

I'll leave it to @BillWagner to advise about which language version to target.

**Comment by BillWagner** [comment]
Hi @mdell-seradex 

Using a C# language version that's higher than your TFM is unsupported. It's complicated to reason about which features work in what conditions. You need to have knowledge of the lowered code for a language feature, and any runtime dependencies. If you choose to create your own alternative implementation for a runtime dependency, you need to make sure you satisfy the requirements correctly. 

Specifically, for raw string literals, they work because that feature doesn't have runtime dependencies.

But again, it's an unsupported scenario.

**Comment by mdell-seradex** [comment]
@BillWagner, thank you for that.  I find it strange then that @CyrusNajmabadi would recommend it, and even say that they do it `on roslyn, which still targets .net framework` https://github.com/dotnet/roslyn/issues/73661#issuecomment-2130413984
I was saying that is sounds like an unsupported and not recommended scenario, which you have confirmed.
I appreciate that.

**Comment by BillWagner** [comment]
@mdell-seradex 

@CyrusNajmabadi is also correct. With care, it works for some language features. It's used in roslyn, and in other multi-targeting scenarios, like Visual Studio extensions. 

It requires discipline and care. That's why it's unsupported.

**Comment by mdell-seradex** [comment]
@BillWagner, I understand, but it sounds a bit dangerous for a regular group of developers.  The discipline and care is likely the bigger issue.
I would never have expected the answer to my virtual indentation request to have been to use Visual Studio in an unsupported way, without that actually being said.  Instead I would have expected the answer to be that support will not be added for it.
I do not feel that I can recommend this as an option to my company.

**Comment by CyrusNajmabadi** [comment]
> thank you for that. I find it strange then that @CyrusNajmabadi would recommend it, and even say that they do it on roslyn, which still targets .net framework https://github.com/dotnet/roslyn/issues/73661#issuecomment-2130413984

I said that it's what we do.  And that raw string literals have no runtime dependency at all and should work on any framework.  I also stated, clearly:

> You do you. Fundamentally, you make the choices here about yoru codebase. I'm simply explaining how these features work and what your options are. If you choose to not use them, c'est la vie.

**Comment by mdell-seradex** [comment]
He did say that `The compiler will error if you try to use an unsupported feature`, but based on the documentation, this does not seem to entirely be true, especially if discipline and care must be taken.  Anyway, I have "hijacked" this discussion enough.

**Comment by CyrusNajmabadi** [comment]
> He did say that The compiler will error if you try to use an unsupported feature, but based on the documentation, this does not seem to entirely be true,

Can you give an example of where thsi is not the case?  If there is a runtime dependency we check the runtime you're compiling for to ensure it can support that dependency.

**Comment by mdell-seradex** [comment]
My statement is based on the documentation and what @BillWagner had said.  If he and the documentation are wrong, then that should be addressed.
I cannot risk not finding issues, only for others on my team to find them.

**Comment by CyrusNajmabadi** [comment]
> I cannot risk not finding issues, only for others on my team to find them.

There is no version of C# or .Net where this is not the case.  None of our products are risk-free.  And you may encounter any number of issues or incompatibilities no matter what you pick.

**Comment by mdell-seradex** [comment]
While that is true, there are different degrees, and supported and unsupported scenarios.
In a supported scenario, the risk is relatively low, and in many cases Microsoft will address the issue.  This is not true of unsupported scenarios.
That recommendation is unsupported which adds extra risk and concern.  If it were just me, that that may be different.

**Pull request merged**
